### ISSUE TITLE
Fine-grained mode doesn't properly respect `--follow-imports` settings when producing error messages

### ISSUE THREAD

**Issue by msullivan** [python/mypy#4798]
This test fails:
```
[case testSkipButDontIgnore]
# cmd: mypy a.py c.py
# flags: --follow-imports=skip
[file a.py]
import b
from c import x
[file b.py]
[file c.py]
x = 1
[file c.py.2]
x = '2'
[out]
==
```
producing
```
a.py:1: error: Cannot find module named 'b'
a.py:1: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
```
in the second iteration.
There will be similar problems with incorrect diagnostics with `--follow-imports=error`.

The root of the problem is that the `verify_dependencies` function used to produce missing modules messages during incremental updates operates solely in terms of the module graph, but producing the correct diagnostics here requires consulting the file system.

None of this is a problem if you use `--ignored-unused-imports`, of course.

**Comment by JukkaL** [comment]
Currently we only properly support `--follow-imports=skip --ignore-missing-imports` in fine-grained mode (I assume that you meant `--ignore-missing-imports` instead of `--ignored-unused-imports`).
